Added an API to allow oversize protos when using C++ extension in Python

keveman opened the pull request.

haberman commented on python/google/protobuf/internal/oversize_protos_test.py:
```diff
@@ -0,0 +1,106 @@
+
+@unittest.skipIf(api_implementation.Type() != 'cpp',
+                 'explicit tests of the C++ implementation')
+class OversizeProtosTest(unittest.TestCase):
```
Please add this class in an existing file, like message_test.py. Our open-sourcing process makes adding new files a big pain, and there isn't much code here.

xfxyjwf commented:
@haberman @anandolee

grpc-jenkins commented:
Can one of the admins verify this patch?

haberman commented on python/google/protobuf/internal/oversize_protos_test.py:
```diff
@@ -0,0 +1,106 @@
+  def testAssertOversizeProto(self):
+    from google.protobuf.pyext._message import SetAllowOversizeProtos
+    SetAllowOversizeProtos(False)
+    proto_cls = proto_builder.MakeProtoClassFromFileDescriptor(
```
Instead please use an interface like DescriptorPool or Descriptor. The proto_builder interface is kind of a random interface with few users that I'd like to deprecate if possible.

haberman commented:
Change looks great in general! I just had a few comments.

haberman commented:
ok to test

haberman commented:
Whoops, never mind that Jenkins failure, looks like we still have some bugs to iron out in that.

haberman commented:
retest this please

keveman commented:
Whoops, looks like we lost some of your comments because I squashed my commits into one. But anyways, I addressed both of your comments : 

> Instead please use an interface like DescriptorPool or Descriptor. The proto_builder interface is kind of a random interface with few users that I'd like to deprecate if possible.

> Please add this class in an existing file, like message_test.py. Our open-sourcing process makes adding new files a big pain, and there isn't much code here.

PTAL.

haberman commented:
LGTM, thanks. Will just wait for the tests to complete before merging.

haberman commented:
retest this please

haberman commented:
(that comment was for Jenkins btw. :)

haberman commented:
Congratulations, you are the first beneficiary of a sub-10 minute Jenkins build!

I'll merge this now, since I don't foresee any platform-specific breakage on this.
